velox/vector/BaseVector.h

```diff
@@ -863,6 +863,18 @@ class BaseVector {
     storageByteCount_ = std::nullopt;
   }
 
+  /// Slice a buffer with specific type.
+  /// For boolean type and if the 'offset' is not multiple of 8, return a
+  /// shifted copy, new buffer is allocated from 'pool'.
+  /// Otherwise return a BufferView into the original buffer (with shared
+  /// ownership of original buffer).
+  static BufferPtr sliceBuffer(
+      const Type& type,
```

Comment:
I assume there are some restrictions on 'type' here. Are only primitive physical types supported? Can you document this restriction? Do we have a check in the implementation? If not, let's add it.

Reply:
> Are only primitive physical types supported?

Primitive types and OPAQUE type are supported.

> Can you document this restriction? Do we have a check in the implementation?

Doc is updated, there is already a check in `typeSize` function.